update_conda_lock/update_lock.py

```diff
@@ -0,0 +1,257 @@
+    try:
+        with open(path, 'r') as f:
+            old_lock = f.read()
+            if old_lock == new_lock:
+                print(path + ' is up to date.')
+                print()
+                return False
+    except FileNotFoundError:
+        print(path + ' doesn\'t exist; it will be created.')
+    with open(path, 'w') as f:
+        f.write(new_lock)
+    print(path + ' has been updated successfully!')
+    print()
+    return True
+
+def analyze_pip_requirement(requirement, analyzed_file_dir):
+    path_match = match(r'-r (file:)?(.*)', requirement)
+    if path_match is None:
+        return [ requirement ]
+    else:  # `requirement` includes some additional `requirements.txt` file
+        # `-r PATH` is relative to the environment file
+        req_path = join(analyzed_file_dir, path_match.group(2))
+        print('Found additional pip requirements file: ' + req_path)
+        with open(req_path, 'r') as f:
+            file_requirements = []
+            for req_line in f.readlines():
+                file_requirements.extend(
+                        analyze_pip_requirement(req_line, dirname(req_path))
+                )
+            return file_requirements
+
+def get_all_pip_dependencies(pip_dependencies, analyzed_file_dir):
+    all_pip_dependencies = []
+    for pip_dependency in pip_dependencies:
+        all_pip_dependencies.extend(
+                analyze_pip_requirement(pip_dependency, analyzed_file_dir)
+        )
+    return all_pip_dependencies
+
+def extract_pip_dependencies(env_yml_path):
+    with open(env_yml_path, 'r') as f:
+        env_yml = yaml.load(f.read())
+
+    pip_dependencies = None
+    for dependency in env_yml['dependencies']:
+        # `- pip:` line becomes a dict-like object with `pip` key after parsing
+        if isinstance(dependency, dict) and 'pip' in dependency.keys():
+            # `pip:` key is replaced with `pip` package to have it installed
+            # even when there was only `pip:` key in `environment.yml`
+            env_yml['dependencies'].remove(dependency)
+            env_yml['dependencies'].append('pip')
+            env_yml_pip_dependencies = list(dependency['pip'])
+
+            # Save `environment.yml` without pip requirements
+            env_yml_path = 'bot-env.yml'
+            with open(env_yml_path, 'w') as f:
+                yaml.dump(env_yml, f)
+    return (env_yml['name'], env_yml_path, env_yml_pip_dependencies)
+
+
+def get_local_pip_dependencies(pip_dependencies, root_dir):
+    local_pip_dependencies = []
+    local_pip_deps_names = []
+    for dependency in pip_dependencies:
+        dependency = dependency.strip()
+        # Handle comments
+        if dependency.startswith('#'):
+            continue
+        only_dependency = sub(r'(.*)\s+#.*$', r'\1', dependency)
+
+        # Find core of the dependency line without version etc.
+        core_match = search(r'(^|\s)([^\s-][^\s=<>~!;]+)', only_dependency)
+        if core_match is not None:
```

Comment:
The indenting here is getting pretty extreme.
```suggestion
        if core_match is None:
            continue
```

Reply:
Thanks for pointing that out.